src/test/java/org/dataloader/DataLoaderTest.java

```diff
@@ -665,14 +727,19 @@ public void should_work_with_duplicate_keys_when_caching_disabled() throws Execu
         assertThat(future1.get(), equalTo("A"));
         assertThat(future2.get(), equalTo("B"));
         assertThat(future3.get(), equalTo("A"));
-        assertThat(loadCalls, equalTo(singletonList(asList("A", "B", "A"))));
+        if (factory instanceof ListDataLoaderFactory) {
```

Comment:
I was a bit iffy on this - wasn't sure if I should split this out into a separate test but it's 95% the same code.